### PR TITLE
enable cross version tests for diviner

### PR DESCRIPTION
**Pull request by BenWilson2**
Signed-off-by: Ben Wilson <benjamin.wilson@databricks.com>

## What changes are proposed in this pull request?

Enable cross version tests for diviner

## How is this patch tested?

cross version tests on Github Actions

## Does this PR change the documentation?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Make sure the changed pages / sections render correctly by following the steps below.

1. Check the status of the `ci/circleci: build_doc` check. If it's successful, proceed to the
   next step, otherwise fix it.
2. Click `Details` on the right to open the job page of CircleCI.
3. Click the `Artifacts` tab.
4. Click `docs/build/html/index.html`.
5. Find the changed pages / sections and make sure they render correctly.

### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

(Details in 1-2 sentences. You can just refer to another PR with a description if this PR is part of a larger change.)

### What component(s), interfaces, languages, and integrations does this PR affect?
Components 
- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface 
- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language 
- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations
- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>
### How should the PR be classified in the release notes? Choose one:

- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

File: mlflow/ml-package-versions.yml [hunk 1]
```diff
@@ -378,3 +378,18 @@ pmdarima:
     requirements: ["prophet"]
     run: |
       pytest tests/pmdarima/test_pmdarima_model_export.py --large
+
+diviner:
+  package_info:
+    pip_release: "diviner"
+    install_dev: |
+      if [ ! -d "$CACHE_DIR" ] || [ -z $(find $CACHE_DIR -type f -name "diviner-*.whl") ]; then
+        pip wheel --no-deps --wheel-dir $CACHE_DIR git+https://github.com/databricks/diviner.git
```
Comment (harupy) [inline mlflow/ml-package-versions.yml:387]: How long does it take for `pip git+https://github.com/databricks/diviner.git` to complete?

Comment (harupy) [reply mlflow/ml-package-versions.yml:387]: If it's short, just running `pip install git+https://github.com/databricks/diviner.git` is ok (and simpler).

Comment (BenWilson2) [reply mlflow/ml-package-versions.yml:387]: It has to build build (not install directly) prophet, statsmodels, and pmdarima. On my local machine it's a few minutes...

Comment (harupy) [reply mlflow/ml-package-versions.yml:387]: Let's run this step twice and see how much this speeds up build.

Comment (harupy) [reply mlflow/ml-package-versions.yml:387]: The current code only caches for a `diviner` wheel, not other package wheels.

**Review by harupy — APPROVED** [review]
LGTM! https://github.com/mlflow/mlflow/pull/5592/files#r842193381 is not a blocker. I'm fine as long as the `install_dev` step works :)

File: mlflow/ml-package-versions.yml (hunk 1 above)
Comment (BenWilson2) [reply mlflow/ml-package-versions.yml:387]: ahhhhhh that makes sense! Yeah I think the entire repo is probably < 10KB :) 
I'll do the direct import in a few minutes!

Comment (BenWilson2) [reply mlflow/ml-package-versions.yml]: updated!

**Pull request merged**